airflow/contrib/operators/qubole_check_operator.py

```diff
@@ -0,0 +1,334 @@
+            :db_table: name of the db table
+            :where_clause: where clause, if any
+            :parallelism: number of parallel db connections to use for extracting data
+            :extract_query: SQL query to extract data from db. $CONDITIONS must be part
+                of the where clause.
+            :boundary_query: Query to be used get range of row IDs to be extracted
+            :split_column: Column used as row ID to split data into ranges (mode 2)
+
+    .. note:: Following fields are template-supported : ``query``, ``script_location``,
+        ``sub_command``, ``script``, ``files``, ``archives``, ``program``, ``cmdline``,
+        ``sql``, ``where_clause``, ``extract_query``, ``boundary_query``, ``macros``,
+        ``tags``, ``name``, ``parameters``, ``dbtap_id``, ``hive_table``, ``db_table``,
+        ``split_column``, ``db_update_keys``, ``export_dir``, ``partition_spec``,
+        ``pass_value``. You can also use ``.txt`` files for template driven use cases.
+    """
+
+    template_fields = QuboleOperator.template_fields + ValueCheckOperator.template_fields
+    template_ext = QuboleOperator.template_ext
+    ui_fgcolor = '#000'
+
+    @apply_defaults
+    def __init__(self, pass_value, tolerance=None,
+                 qubole_conn_id="qubole_default", *args, **kwargs):
+
+        sql = get_sql_from_qbol_cmd(kwargs)
+        super(QuboleValueCheckOperator, self).__init__(
+            qubole_conn_id=qubole_conn_id,
+            sql=sql, pass_value=pass_value, tolerance=tolerance,
+            *args, **kwargs)
+
+        self.on_failure_callback = QuboleCheckHook.handle_failure_retry
+        self.on_retry_callback = QuboleCheckHook.handle_failure_retry
+        self.hook = QuboleCheckHook(context=None, *self.args, **self.kwargs)
+
+    def execute(self, context=None):
+        try:
+            self.hook = QuboleCheckHook(context=context, *self.args, **self.kwargs)
+            super(QuboleValueCheckOperator, self).execute(context=context)
+        except AirflowException as e:
+            handle_airflow_exception(e, self.get_db_hook())
+
+    def get_db_hook(self):
+        return self.hook
+
+    def __getattribute__(self, name):
+        if name in QuboleValueCheckOperator.template_fields:
+            if name in self.kwargs:
+                return self.kwargs[name]
+            else:
+                return ''
+        else:
+            return object.__getattribute__(self, name)
+
+    def __setattr__(self, name, value):
+        if name in QuboleValueCheckOperator.template_fields:
+            self.kwargs[name] = value
+        else:
+            object.__setattr__(self, name, value)
+
+
+def get_sql_from_qbol_cmd(params):
+    sql = ''
+    if 'query' in params:
+        sql = params['query']
+    elif 'sql' in params:
+        sql = params['sql']
+    return sql
+
+
+def handle_airflow_exception(airflow_exception, hook):
+    cmd = hook.cmd
+    if cmd is not None:
+        if cmd.status == 'done':
```

Comment:
better to use sdk given method, i.e. https://github.com/qubole/qds-sdk-py/blob/master/qds_sdk/commands.py#L55